src/torrent.cpp

```diff
@@ -10497,9 +10496,8 @@ bool is_downloading_state(int const st)
 	// adds a piece
 	void torrent::verify_piece(piece_index_t const piece)
 	{
-//		picker().mark_as_checking(piece);
-
 		TORRENT_ASSERT(m_storage);
+		TORRENT_ASSERT(!m_picker->is_hashing(piece));
```

Comment:
this may be controversial, pending your response to a comment in the main PR. If this turns out to be a correct assert (i.e. it cannot happen), the current `hashing` counter could be turned into one bit